File: python-version/LC594/find_LHS.py
class Solution:
    @staticmethod
    def findLHS(nums: list[int]) -> int:
        ans = 0
        mp = dict()
        for num in nums:
            mp[num] = mp.get(num, 0) + 1

        for k, v in mp.items():
            if mp.__contains__(k+1):
                ans = max(ans, v + mp.get(k+1))
        return ans

File: go-version/leetcode/LC594/find_LHS.go
package LC594

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/6/30 下午10:54
 * @FilePath: find_LHS
 * @Description:
 */

func findLHS(nums []int) int {
	var ans int
	mp := make(map[int]int)
	for _, num := range nums {
		mp[num]++
	}

	for val, cnt := range mp {
		if cntNext, ok := mp[val+1]; ok {
			ans = max(ans, cnt+cntNext)
		}
	}
	return ans
}
